compiler/modules/types/list.bal

```diff
@@ -284,21 +291,19 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, Conju
         // SemType d1 = diff(s[1], t[1]);
         // return !isEmpty(cx, d1) &&  tupleInhabited(cx, [s[0], d1], neg.rest);
         // We can generalize this to tuples of arbitrary length.
-        int maxInitialLength = members.initial.length();
-        if maxInitialLength != 0 {
```

Comment:
Why isn't this check needed now?

Reply:
As I remember, this was added to protect the call to `fixedArrayGet` in line:292, it was replaced by a call to `listMemberAt`.